bigquery/google/cloud/bigquery/job.py

```diff
@@ -702,6 +707,25 @@ def __init__(self, job_type):
         self._job_type = job_type
         self._properties = {job_type: {}}
 
+    @property
+    def labels(self):
+        """Dict[str, str]: Labels for the job.
+
+        This method always returns a dict. To change a job's labels,
+        modify the dict, then call ``Client.update_job``. To delete a
+        label, set its value to :data:`None` before updating.
+
+        Raises:
+            ValueError: If ``value`` type is invalid.
+        """
+        return self._properties.get('labels', {})
```

Comment:
Per the docstring, you should be able to modify the dictionary in-place. I think that use-case won't work when the dictionary is originally not present. Maybe explicitly check for the `labels` key and set to `{}` if not present instead of using `.get()`?

Reply:
076be15 Updates *all* `labels` getters to use `setdefault`, and adds tests for updating in place.